### PR TITLE
refactor(boards): improve board index building; board-assets and ScrollList boards

### PR DESCRIPTION
**Pull request by plisovyi**
I did some renaming, moved all ScrollList boards to shared items creation, and fixed typos.

### CODE REVIEW

File: build-board-indexes.js [hunk 1]
```diff
@@ -4,8 +4,9 @@ const fs = require('fs');
 glob('**/*.board.tsx', (error, boards) => {
     if (error) {
         // eslint-disable-next-line no-console
-        console.log(error);
+        console.error(error);
```
Comment (YardenPorat) [inline build-board-indexes.js:7]: shouldn't it throw?

File: packages/components/src/auto-complete/auto-complete.board.tsx [hunk 2]
```diff
@@ -1,17 +1,13 @@
-import React from 'react';
 import { createBoard } from '@wixc3/react-board';
-import { ZeeRoot } from '../simulation-mixins/mixin-zee-root';
+import React from 'react';
+import type { ItemData } from '../board-assets';
+import { createItems, ItemRenderer } from '../board-assets';
+import { mixinProjectThemes } from '../board-mixins/mixin-project-themes';
+import { ZeeRoot } from '../board-mixins/mixin-zee-root';
 import { AutoComplete } from './auto-complete';
```
Comment (YardenPorat) [inline packages/components/src/auto-complete/auto-complete.board.tsx:7]: we like having an import order of:
1. 3rd party deps
2. monorepo packages
3. relative
4. style
5. type

this is only a recommendation

**Review by YardenPorat — APPROVED** [review]
approved with comments

File: build-board-indexes.js (hunk 1 above)
Comment (plisovyi) [reply build-board-indexes.js:7]: It's a design decision, I guess. Here, it'd run on every board file, and on the throw, the whole script would exit with an error. While with `console.error`, a message would go out, but processing of boards would still proceed, and this isn't stopping the npm script execution flow. 
Or something 🤷🏼

**Pull request merged**